### ISSUE TITLE
MDCSnackBar are not read in voice over

### ISSUE THREAD

**Issue by gambard** [material-components/material-components-ios#1237]
### Overview
When displaying a transient snackbar with voice over on, the snackbar is very briefly read then the focus is switched to another element, preventing the snackbar to be fully read.

### Reproduction steps
Example at crbug.com/698719

### Version number
22.0.0

### Related issues
I think this is caused by #1214

**Comment by Arcank** [comment]
Ian, Sam, could you please triage this issue?

**Comment by samnm** [comment]
This is not an easy fix unfortunately. #1214 returns MDCSnackbar voiceover functionality to how it was intended to work by [posting a notification with the snackbar contents](https://github.com/material-components/material-components-ios/blob/develop/components/Snackbar/src/MDCSnackbarManager.m#L219), instead of posting a notification pointing to the snackbar itself. As far as I can tell since Chrome iOS posts the snackbar while the share sheet is dismissing, the voiceover notification gets overridden by the original view controller being brought back to the foreground. 

In order to fix this I think we'd have to change the functionality of snackbar voiceovers so that modal snackbars get full voiceover focus.

**Comment by samnm** [comment]
One (kind of hacky) solution I think might work would be to wrap the UIAccessibilityPostNotification call inside a `dispatch_async(dispatch_get_main_queue(), ^{ ... });` so as to have the notification posted after the view controller transitions have finished.

**Comment by ianegordon** [comment]
FYI: Internal implementation
cl/154072838

**Comment by ianegordon** [comment]
It looks like the internal fix was Rolled back.
The current internal implementation still has VO notifications overlapping.

Do we have a final implementation or minimally a strategy?

**Comment by samnm** [comment]
@gambard + @Arcank Do you happen to have any thoughts regarding my earlier comments? I don't think we can fix this in MDC so I'm curious to hear what you think.

**Comment by gambard** [comment]
I thought your solution was for MDC. Dispatching it async does not work as the view controller get the focus 0.5s after. Adding a 1s delay works but this is not a great user experience.

**Comment by samnm** [comment]
I think this is simply an unfortunate side effect of the modality of Snackbars and the way voiceover works on iOS. My suggestion would be to post the snackbar message after a delay or use something other than Snackbars.

I'm going to close this issue because as far as I am concerned there is no reasonable fix, but I'm happy to hear any and all suggestions if people have ideas.
